compiler/x/codegen/X86SystemLinkage.cpp

```diff
@@ -429,7 +429,7 @@ TR::X86SystemLinkage::createPrologue(TR::Instruction *cursor)
 
    const TR::X86LinkageProperties &properties = getProperties();
 
-   const uint32_t outgoingArgSize = cg()->getLargestOutgoingArgSize();
+   const uint32_t outgoingArgSize = (cg()->isLeafMethod()) ? 0 : cg()->getLargestOutgoingArgSize();
```

Comment:
heh, good point. I'll review that code to see if I can reconstruct why I decided to solve it here.